lib/faraday/adapter/test.rb

```diff
@@ -239,26 +233,19 @@ def configure
         yield(stubs)
       end
 
+      # @param env [Faraday::Env]
       def call(env)
         super
-        host = env[:url].host
-        normalized_path = Faraday::Utils.normalize_path(env[:url])
-        params_encoder = env.request.params_encoder ||
-                         Faraday::Utils.default_params_encoder
-
-        stub, meta = stubs.match(env[:method], host, normalized_path,
-                                 env.request_headers, env[:body])
+        stub, meta = stubs.match(env)
 
         unless stub
+          normalized_path = Faraday::Utils.normalize_path(env[:url])
           raise Stubs::NotFound, "no stubbed request for #{env[:method]} "\
                                  "#{normalized_path} #{env[:body]}"
         end
 
-        env[:params] = if (query = env[:url].query)
-                         params_encoder.decode(query)
-                       else
-                         {}
-                       end
+        params_encoder = env.request.params_encoder || Faraday::Utils.default_params_encoder
+        env[:params] = params_encoder.decode(env[:url].query) || {}
```

Comment:
Mmmmh I really like how much simpler the `matches?` methods signature are now, thanks to the `env` being passed, but I notice there's a lot a of repeated code between here and there. These two lines for example, and line 242 as well.

I wonder if instead of reusing `env` it might make sense to introduce a new `Struct`, internal and only used in the Test adapter, that holds this information so we only need to calculate it once. What do you think?

Reply:
Thank you for your quick review! It makes sense 👍 I'll try it.

Reply:
I tried to avoid similar codes with 91689f0. How about the change?